test/support/utils_sycl.h

```diff
@@ -132,28 +132,106 @@ make_new_policy(_Policy&& __policy)
 // create the queue with custom asynchronous exceptions handler
 static auto my_queue = sycl::queue(default_selector, async_handler);
 
-// Invoke op(policy,rest...) for each possible policy.
+// check fp16 and fp64 support
+template<typename T>
+bool has_type_support(const sycl::device& device)
```

Comment:
May be we should have 3 functions with SFINAE that will check same type ?

Reply:
I specialized `has_type_support` for `double` and `sycl::half` for simplicity instead of involving SFINAE. If you believe using SFINAE is better, then feel free to reopen the issue.